### PR TITLE
[AD-1014] Developer Guide.

### PR DESCRIPTION
**Pull request by birschick-bq**
### Summary

[AD-1014] Developer Guide.

### Description

- Add links for developers to the GETTING_STARTED guide.
- Add documentation of schema caching behaviour and access requirements.

### Related Issue

https://bitquill.atlassian.net/browse/AD-1014

### Additional Reviewers
@affonsoBQ
@alinaliBQ
@andiem-bq
@alexey-temnikov
@birschick-bq

### CODE REVIEW

File: src/markdown/schema/schema-caching.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+When a connection is made to an Amazon DocumentDB database, the Amazon DocumentDB JDBC driver 
+checks for a previously cached version of the mapped schema. If a previous version exists, 
+the latest version of the cached schema is read and used for all further interaction with the database.
+
+If a previously cached version does not exist, the process of [schema discover](schema-discovery.md) is automatically
+started on all the accessible collections in the database. The discovery process uses the properties 
+`scanMethod` (default `random`), and `scanLimit` (default `1000`) when sampling documents from the database.
+At the end of the discovery process, the resulting schema mapping is written to the cache using the name
+associated with the property `schemaName` (default `_default`).
```
Comment (alinaliBQ) [inline src/markdown/schema/schema-caching.md:13]: Add link to instructions on how the user can configure those properties to a non-default value?
E.g., "For how to configure `scanMethod`, `scanLimit`, and `schemaName`, check [this page](url) to learn more about the values"

File: src/markdown/schema/schema-caching.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+If some reason the resulting schema cannot be saved to the cache, the resulting schema will still be used
```
Comment (alinaliBQ) [inline src/markdown/schema/schema-caching.md:15]:
```suggestion
If for some reason the resulting schema cannot be saved to the cache, the resulting schema will still be used
```

File: src/markdown/schema/schema-caching.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
+# Schema Caching
+
+## Schema Caching Behaviour
+
+When a connection is made to an Amazon DocumentDB database, the Amazon DocumentDB JDBC driver 
+checks for a previously cached version of the mapped schema. If a previous version exists, 
+the latest version of the cached schema is read and used for all further interaction with the database.
+
+If a previously cached version does not exist, the process of [schema discover](schema-discovery.md) is automatically
```
Comment (andiemontoyeah) [inline src/markdown/schema/schema-caching.md:9]:
```suggestion
If a previously cached version does not exist, the process of [schema discovery](schema-discovery.md) is automatically

```

**Review by andiemontoyeah — COMMENTED** [review]
lgtm i think!

**Review by andiemontoyeah — APPROVED** [review]
LGTM!

**Pull request merged**